cluster/src/dunit/scala/io/snappydata/cluster/ClusterMgrDUnitTest.scala

```diff
@@ -44,11 +44,15 @@ class ClusterMgrDUnitTest(s: String) extends ClusterManagerTestBase(s) {
     // connect with this new lead.
     // In this case servers are already running and a lead comes
     // and join
-    vm3.invoke(getClass, "startSnappyLead", startArgs)
-    vm3.invoke(getClass, "startSparkJob")
-    vm3.invoke(getClass, "startGemJob")
-    vm3.invoke(getClass, "stopSpark")
-    ClusterManagerTestBase.startSnappyLead(ClusterManagerTestBase.locatorPort, bootProps)
+    try {
+      vm3.invoke(getClass, "stopAny")
```

Comment:
This moves the stopAny and startSnappyLead calls in the finally block. Otherwise if there is a failure in this test then vm3 remains the lead while current one is removed, while subsequent tests will try to make the current one as lead again and hang on the lead lock which is acquired by vm3.